hasty/export_job.py

```diff
@@ -0,0 +1,116 @@
+from typing import List, Union
+import os
+import urllib.request
+
+from .hasty_object import HastyObject
+from .requester import Requester
+
+
+class ExportJob(HastyObject):
+    endpoint = '/v1/projects/{project_id}/exports'
+    endpoint_export = '/v1/projects/{project_id}/exports/{export_id}'
+
+    def __repr__(self):
+        return self.get__repr__(OrderedDict({"id": self._id, "status": self._status}))
+
+    @property
+    def id(self):
+        """
+        :type: string
+        """
+        return self._id
+
+    @property
+    def status(self):
+        """
+        :type: string
+        """
+        return self._status
+
+    @property
+    def project_id(self):
+        """
+        :type: string
+        """
+        return self._project_id
+
+    @property
+    def started_on(self):
+        """
+        :type: string
+        """
+        return self._started_on
+
+    @property
+    def completed_on(self):
+        """
+        :type: string
+        """
+        return self._completed_on
+
+    @property
+    def meta(self):
+        return self._meta
+
+    def _init_properties(self):
+        self._id = None
+        self._status = None
+        self._project_id = None
+        self._started_on = None
+        self._completed_on = None
+        self._meta = None
+
+    def _set_prop_values(self, data):
+        if "id" in data:
+            self._id = data["id"]
+        if "status" in data:
+            self._status = data["status"]
+        if "started_on" in data:
+            self._started_on = data["started_on"]
+        if "completed_on" in data:
+            self._completed_on = data["completed_on"]
+        if "meta" in data:
+            self._meta = data["meta"]
+        if "project_id" in data:
+            self._project_id = data["project_id"]
+
+    @staticmethod
+    def _create(requester: Requester, project_id: str, name: str, export_format: str,
+                dataset: List[str], image_status: List[str], sign_urls: bool,
+                semantic_format: str, labels_order: List[str]):
+        res = requester.post(ExportJob.endpoint.format(project_id=project_id),
+                             json_data={"export_name": name,
+                                        "format": export_format,
+                                        "dataset_id": dataset,
+                                        "image_status": image_status,
+                                        "sign_urls": sign_urls,
+                                        "semantic_format": semantic_format,
+                                        "labels_order": labels_order})
+        return ExportJob(requester, res, {"project_id": project_id})
+
+    def check_status(self):
+        """
+        Checks and returns the status of the export job.
+        """
+        res = self._requester.get(ExportJob.endpoint_export.format(project_id=self._project_id, export_id=self._id))
+        self._set_prop_values(res)
+        return self._status
```

Comment:
Should we cache this result for like 10 seconds or something? I'm wary of having this in a loop.

Reply:
Fixed